src/python/dxpy/scripts/dx_build_app.py

```diff
@@ -211,6 +211,69 @@ def parse_destination(dest_str):
     return try_call(resolve_path, dest_str)
 
 
+def _check_suggestions(src_dir, publish=False):
+    """
+    Examines the specified dxapp.json file and warns about any
+    violations of suggestions guidelines.
+    """
+    dxapp_json_filename = os.path.join(src_dir, "dxapp.json")
+    app_spec = json.load(open(dxapp_json_filename))
+    if 'inputSpec' in app_spec:
+        for input_field in app_spec['inputSpec']:
```

Comment:
Here and elsewhere, you can reduce the nesting level a bit by using "get" with an appropriate default value, e.g.:

```
for input_field in app_spec.get('inputSpec', []):
    for suggestion in input_field.get('suggestions', []):
        if 'project' in suggestion:
```